analyzers/tests/SonarAnalyzer.UnitTest/Extensions/SyntaxNodeExtensionsTest.cs

```diff
@@ -207,7 +254,41 @@ static int InnerLocalFunction(int arg)
         }
 
         [TestMethod]
-        public void CreateCfg_UndefinedSymbol_ReturnsCfg()
+        public void CreateCfg_NestingChain_VB()
+        {
+            var code = @"
+Public Class Sample
+    Public Sub Main(Values() As Integer)
+        Dim OuterMultiLineSub As Action =
+            Sub()
+                Dim OuterSingleLineFunc As Func(Of Func(Of Integer, Integer)) =
+                        Function() Function(NestedMultiLineFuncArg As Integer)
+                                        Dim Lst = From Value
+                                                  In Values
+                                                  Select New Lazy(Of Integer)(Function()
+                                                                                  Dim InnerSingleLineSub = Sub() Value.ToString()
+                                                                              End Function)
+                                    End Function
+            End Sub
```

Comment:
Good luck with cracking this madness 🤣

It's similar to the C# above - it just tries to combine all possible nesting and makes sure that the correct CFG is returned.